apps/server/src/shared/testing/test-xApiKey-client.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { Controller, Delete, ExecutionContext, Get, Headers, HttpStatus, INestApplication, Post } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import { ObjectId } from 'bson';
+import { AuthGuard } from '@nestjs/passport';
+import { TestXApiKeyClient } from './test-xApiKey-client';
+
+@Controller('')
+class TestController {
+	@Delete(':id')
+	async delete(@Headers('X-API-KEY') authorization: string) {
+		return Promise.resolve({ method: 'delete', authorization });
+	}
+
+	@Post()
+	async post(@Headers('X-API-KEY') authorization: string) {
+		return Promise.resolve({ method: 'post', authorization });
+	}
+
+	@Get(':id')
+	async get(@Headers('X-API-KEY') authorization: string) {
+		return Promise.resolve({ method: 'get', authorization });
+	}
+}
+
+describe(TestXApiKeyClient.name, () => {
+	describe('when test request instance exists', () => {
+		let app: INestApplication;
+		const API_KEY = '1ab2c3d4e5f61ab2c3d4e5f6';
+
+		beforeAll(async () => {
+			const moduleFixture = await Test.createTestingModule({
+				controllers: [TestController],
+			})
+				.overrideGuard(AuthGuard('api-key'))
```

Comment:
Please remove this override. The goal of this helper should be that it is work without. I think it can be done with the test-api-client, it exists no need to add a additional helper.

Reply:
@WojciechGrancow or
I guess waiting for answer on comment https://github.com/hpi-schul-cloud/schulcloud-server/pull/4533#discussion_r1407526275